packages/mds-ingest-service/repository/index.ts

```diff
@@ -338,26 +344,27 @@ class IngestReadWriteRepository extends ReadWriteRepository {
       return {
         events: data.map(EventEntityToDomain.mapper()),
         cursor: {
-          next: nextAfterCursor && this.buildCursor({ ...cursor, beforeCursor: null, afterCursor: nextAfterCursor }),
-          prev: nextBeforeCursor && this.buildCursor({ ...cursor, beforeCursor: nextBeforeCursor, afterCursor: null })
+          next:
+            nextAfterCursor &&
+            this.buildCursor<GetVehicleEventsFilterParams>({
```

Comment:
Could perhaps avoid some `this.` calls by doing a destructure of this method when you yoink out `connect`